pkg/domain/app_test.go

```diff
@@ -322,3 +322,29 @@ func TestApplication_SelfValidate(t *testing.T) {
 		})
 	}
 }
+
+func TestHashWithEnv(t *testing.T) {
+	config := ApplicationConfig{
+		BuildConfig: &BuildConfigRuntimeDockerfile{DockerfileName: "Dockerfile"},
+	}
+	require.NoError(t, config.Validate(DeployTypeRuntime))
+
+	testEnv := []*Environment{
+		{Key: "key1", Value: "value1"},
+		{Key: "key2", Value: "value2"},
+	}
+	hash := config.Hash(testEnv)
+
+	testEnv = append(testEnv, &Environment{Key: "key3", Value: "value3"})
+	t.Run("add env", func(t *testing.T) {
+		assert.NotEqual(t, hash, config.Hash(testEnv))
+	})
```

Comment:
subtestの中身はsubtestの中で完結させるようにするといいですね
t.Runに与える名前は、そのsubtestの内容を簡単に説明するものが一般的なはずです
```suggestion
	t.Run("hash should not be equal when env is different", func(t *testing.T) {
		testEnv2 := []*Environment{
			{Key: "key1", Value: "value1"},
			{Key: "key2", Value: "value2"},
			{Key: "key3", Value: "value3"},
		}
		assert.NotEqual(t, hash, config.Hash(testEnv))
	})
```